archive/zchunk.ksy

```diff
@@ -0,0 +1,188 @@
+meta:
+  id: zchunk
+  title: Zchunk
+  file-extension: zck
+  license: CC0-1.0
+  endian: le
+doc-ref: https://github.com/zchunk/zchunk/blob/main/zchunk_format.txt
+seq:
+  - id: lead
+    type: lead
+  - id: header
+    type: header
+    size: lead.len_header.value
+  - id: chunks
+    size: chunk_metadata[_index].len_chunk.value
+    repeat: expr
+    repeat-expr: num_chunks
+instances:
+  num_chunks:
+    value: header.index.num_chunks.value - 1
+    doc: the number of chunks includes the header, so -1
+  chunk_metadata:
+    value: header.index.chunks_metadata
+types:
+  lead:
+    seq:
+      - id: magic
+        contents: [0, 'ZCK1']
+      - id: checksum
+        type: compressed_integer
+      - id: len_header
+        type: compressed_integer
+      - id: lead_checksum
+        size: len_checksum
+    instances:
+      checksum_type:
+        value: checksum.value
+        enum: checksum_types
+      len_checksum:
+        value: |
+            checksum_type == checksum_types::sha1 ? 20 :
+            checksum_type == checksum_types::sha256 ? 32 :
+            checksum_type == checksum_types::sha512 ? 64 :
+            checksum_type == checksum_types::sha512_128 ? 16 :
+            0
+  header:
+    seq:
+      - id: preface
+        type: preface
+      - id: len_index
+        type: compressed_integer
+      - id: index
+        type: index
+        size: len_index.value
+      - id: signatures
+        type: signatures
+  preface:
+    seq:
+      - id: checksum
+        size: _root.lead.len_checksum
+      - id: flags
+        type: compressed_integer
+      - id: compression_type
```

Comment:
I think it may make sense to create an own type for each type over a compressed integer (also, `-affected-by: 135`). I.e. this one should read a compressed integer in a `seq` and create a `value` instance with `enum`. Also may make sense to utilise `kaitai.compress`.